src/components/Code/Code.js

```diff
@@ -0,0 +1,107 @@
+import { isFunction } from 'lodash';
+
+const CopyNotice = styled.div`
+  position: absolute;
+  top: 5px;
+  right: 5px;
+  transition: opacity 300ms ease;
+  opacity: 0;
+  font-size: ${props => props.theme.fontSizes.tiny};
+  color: ${props => props.theme.colors.athensGray};
+`;
+
+const Root = styled.div`
+  position: relative;
+  padding: 10px 15px;
+  background-color: ${props => props.theme.colors.gunpowder};
+  box-sizing: border-box;
+  border-radius: ${props => props.theme.borderRadius.soft};
+  cursor: pointer;
+
+  &:hover ${CopyNotice} {
+    opacity: 1;
+  }
+`;
+
+const Content = styled.div`
+  flex-grow: 1;
+  overflow: auto;
+  color: ${props => props.theme.colors.athensGray};
+`;
+
+const Pre = styled.pre`
+  margin: 10px 0;
+  font-family: ${props => props.theme.fontFamilies.monospace};
+  font-size: ${props => props.theme.fontSizes.small};
+`;
+
+class Code extends Component {
+  static displayName = 'Code';
+
+  state = {
+    isCopying: false,
+  };
+
+  handleCopyClick = () => {
+    // Create a textarea element in which the text of the code will be copied.
+    // Set the value of the textarea, select it to set it as the active element,
+    // then copy the value of that element to the clipboard.
+    // It's 2017 and this is still the best way to copy text on button click...
+    if (isFunction(this.props.onCopy)) {
+      this.props.onCopy();
+    }
+    const selection = document.getSelection().toString();
+    const code = selection !== '' ? this.preNode.textContent : selection;
+    const txtArea = document.createElement('textarea');
+    // Safari doesn't allow for assigning an object literal to `style`.
+    txtArea.class = 'hidden-textarea';
+    // Make sure the code ends in a newline to execute all the commands in multiline code.
+    if (code.charAt(code.length - 1) === '\n') {
+      txtArea.value = code;
+    } else {
+      txtArea.value = `${code}\n`;
+    }
+    document.body.appendChild(txtArea);
+
+    txtArea.select();
+
+    try {
+      document.execCommand('copy');
+    } catch (e) {
+      throw e;
+    }
+    document.body.removeChild(txtArea);
+
+    // show the Copied to clipboard notice temporarily
+    this.setState({ isCopying: true });
+    setTimeout(() => {
+      this.setState({ isCopying: false });
+    }, 1000);
+  };
+
+  render() {
+    const { children } = this.props;
+    const { isCopying } = this.state;
+
+    const copyText = isCopying ? 'Copied to clipboard' : 'Copy to clipboard';
+    const copyIcon = isCopying ? 'files-o' : 'check';
+
+    return (
+      <Root onClick={this.handleCopyClick}>
+        <Content>
+          <Pre innerRef={e => (this.preNode = e)}>
+            {isFunction(children) ? this.props.children() : this.props.children}
+          </Pre>
+        </Content>
+
+        <CopyNotice>
+          <i className={`fa fa-${copyIcon}`} /> {copyText}
+        </CopyNotice>
+      </Root>
+    );
+  }
+}
+
+export default Code;
```

Comment:
`propTypes` (and `defaultProps`) are missing here - we've decided to always add those in this repo, please see any other component around for how to do it.